R/edgeThresh.R

```diff
@@ -0,0 +1,284 @@
+# A value for δ is chosen such that random networks will not result in large subnetworks.
+
+#' Define edge-removal threshold parameter δ for each target subnet-order, from
+#' random permutations of the input EGG object
+#'
+#' \code{THRESH} creates N permutations of the input graph EGG by randomly
+#' choosing 2 edges and swapping their target endpoints (i.e. B in the
+#' edge A → B). Finds the value δ (edge weight) for each random graph such that
+#' the maximum size of any component in the graph is Lmax, and returns the
+#' median of N δ values.
+#'
+#' @section Swapping Edges:
+#'     Bidirectional edges must only be swapped with each other, therefore the
+#'     swapping operation will first be applied to bidirectional edges,
+#'     Q * |E_b| times, where E_b is a bidirectional edge. Since all 4 incident
+#'     nodes in 2 bidirectional edges are receiving endpoints, one node will be
+#'     randomly selected from each edge, and a new edge is formed between the
+#'     selected as well as the remaining vertices.
+#'     Then Q * |E_u| unidirectional edge swaps take place.
+#'     At each unidirectional edge swap, only the receiving endpoints are
+#'     swapped e.g. (A → B, C → D) to (A → D, C → B).
+#'     In both cases, a swap should NOT result in:
+#'     Multiple edges: Possibly result of a bidirectional edge swap.
+#'     Self-edges: e.g. (A ↔ B, C ↔ A) to (A ↔ A, B ↔ C)
+#'     Disconnected components: Considering (A → B, C → D), if A and B are
+#'     in the same component, they should still be in the same component when
+#'     (A → D, C → B) This guarantees that the graph is still connected if
+#'     the starting graph was also connected.
+#'     ### What about (A → B, A → D) to (A → D, A → B)? Will this be
+#'     cosidered a swap? (counted towards Q * |E|) ###
+#'
+#' @section Maintaining Connected Components:
```

Comment:
Since the target outcome is to measure order of the largest connected component, I would use only the largest connected component of the input for the computation in case the input has more than one component. Otherwise, a small component would trigger the threshold even without edge removal.